src/main/scala/it/pagopa/interop/partymanagement/api/impl/PartyApiServiceImpl.scala

```diff
@@ -121,6 +121,76 @@ class PartyApiServiceImpl(
 
   }
 
+  /** Code: 200, Message: successful operation, DataType: Institution
+   * Code: 405, Message: Institution not found, DataType: Problem
+   * Code: 400, Message: Invalid ID supplied, DataType: Problem
+   */
+  override def updateInstitutionById(id: String, institution: Institution)(implicit
+    toEntityMarshallerInstitution: ToEntityMarshaller[Institution],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    contexts: Seq[(String, String)]
+  ): Route = {
+    logger.info(s"Updating institution $id")
+
+    val commander: EntityRef[Command] = getCommander(id)
+      .map(shard => sharding.entityRefFor(PartyPersistentBehavior.TypeKey, shard.toString))
+      .toList
+
+    val updatedInstitution: Future[StatusReply[Party]] = for {
+      uuid      <- id.toFutureUUID
+      found <- commander.ask(ref => GetParty(uuid, ref))
+      party       <- found.toFuture(InstitutionNotFound(id))
+      updatedOrg <- maybeExistingOrg
```

Comment:
This part must be changed.
I suggest creating a function in the Party companion object to extract InstitutionParty, something like:
```  
def extractInstitutionParty(party: Party): Future[InstitutionParty] = {
    party match {
      case p: InstitutionParty => Future.successful(p)
      case p                   => Future.failed(InvalidParty(InstitutionParty.toString, p.toString))
    }
  }
```

Reply:
Once you have obtained the InstituitionParty, you can you the `.copy` function to create the updated version.

Reply:
not using copy because we are replacing the entire object, the only exception are the create and update fields

Reply:
added check to verify if Party is a InstitutionParty